vllm/v1/worker/gpu_model_runner.py

```diff
@@ -3143,6 +3144,22 @@ def may_reinitialize_input_batch(self,
             kv_cache_group.kv_cache_spec.block_size
             for kv_cache_group in kv_cache_config.kv_cache_groups
         ]
+
+        # Generate kernel_block_sizes that matches each block_size
+        # For attention backends that support virtual block splitting,
+        # use the logical block size
+        # For other backends (like Mamba), use 0 (no splitting)
+        kernel_block_sizes = []
+        for kv_cache_group in kv_cache_config.kv_cache_groups:
+            if isinstance(kv_cache_group.kv_cache_spec, AttentionSpec):
+                # This is an attention backend that supports virtual
+                # block splitting.
+                kernel_block_sizes.append(self.cache_config.kernel_block_size)
```

Comment:
Prefer to get the kernel_block_sizes from attention backends. Maybe we will set different attention backend to different layers in the future.